idtoken/validate_test.go

```diff
@@ -27,18 +28,27 @@ const (
 	testAudience = "test-audience"
 )
 
+var (
+	// one second before the token expires
+	beforeExp = func() time.Time { return time.Unix(233431199, 0) }
+	// one second after the token expires
+	afterExp = func() time.Time { return time.Unix(233431201, 0) }
+)
+
 func TestValidateRS256(t *testing.T) {
 	idToken, pk := createRS256JWT(t)
 	tests := []struct {
 		name    string
 		keyID   string
 		n       *big.Int
 		e       int
+		nowFunc func() time.Time
 		wantErr bool
 	}{
-		{name: "works", keyID: keyID, n: pk.N, e: pk.E, wantErr: false},
-		{name: "no matching key", keyID: "5678", n: pk.N, e: pk.E, wantErr: true},
-		{name: "sig does not match", keyID: keyID, n: new(big.Int).SetBytes([]byte("42")), e: 42, wantErr: true},
+		{name: "works", keyID: keyID, n: pk.N, e: pk.E, nowFunc: beforeExp, wantErr: false},
+		{name: "no matching key", keyID: "5678", n: pk.N, e: pk.E, nowFunc: beforeExp, wantErr: true},
+		{name: "sig does not match", keyID: keyID, n: new(big.Int).SetBytes([]byte("42")), e: 42, nowFunc: beforeExp, wantErr: true},
+		{name: "token expired", keyID: keyID, n: pk.N, e: pk.E, nowFunc: afterExp, wantErr: true},
```

Comment:
Optional: might be easier to read these if they were on separate lines.

Reply:
Good point. Done.